Updates Build Instructions to Adjust for Devcontainer Instructions and Remove Stale Infomation (#1179)
rapidsai/cuspatial: docs/source/developer_guide/build.md

2. clone the cuSpatial repo

```shell
conda env create -n cuspatial --file conda/environments/all_cuda-118_arch-x86_64.yaml
```

## Build cuSpatial

### From devcontainer:

Execute `build-cuspatial-cpp -DBUILD_TESTS=ON` to build libcuspatial and tests.
In addition, `build-cuspatial-python` to build cuspatial cython components.

### From bare metal:

Compile libcuspatial (C++), cuspatial (cython) and C++ tests:
```shell
cd $CUSPATIAL_HOME && \
chmod +x ./build.sh && \
./build.sh libcuspatial cuspatial tests
```

> ## Validate installation by running C++ and Python tests

Reviewer: ```suggestion
## Validate Installation with C++ and Python Tests
```

Minor change

> C++ tests locate under `$CUSPATIAL_HOME/cpp/build/gtests`. Python tests locate under
> `$CUSPATIAL_HOME/python/cuspatial/cuspatial/tests`.

Reviewer: ```suggestion
- C++ tests are located within the `$CUSPATIAL_HOME/cpp/build/gtests` directory.
- Python tests are located within the `$CUSPATIAL_HOME/python/cuspatial/cuspatial/tests` directory.
```

> ```note
> Devcontainer users: to manage difference between branches, the build directory is further alternatively placed
> `$CUSPATIAL_HOME/cpp/build/release` and `$CUSPATIAL_HOME/cpp/build/latest`, where `release` places the release build,
> and `latest` is the symlink to the most recent build directory.
> ```

Reviewer: ````suggestion
```note
Dev Container users: 
To manage difference between branches, the build directories are located at
`$CUSPATIAL_HOME/cpp/build/release` and `$CUSPATIAL_HOME/cpp/build/latest`.

Where `release` contains the release build, and `latest` is the symlink to the most recent build directory.
```
````

Not sure if this helps or not, I think it improves clarity but not married to it.

> Execute C++ tests:

Reviewer 2: Please also show how to run all tests. e.g. `ninja -C $CUSPATIAL_HOME/cpp/build/gtests/ test` or appropriate `CTEST` commands (or `make` in case they aren't using Ninja)

```shell
$CUSPATIAL_HOME/cpp/build/gtests/HAUSDORFF_TEST
$CUSPATIAL_HOME/cpp/build/gtests/POINT_IN_POLYGON_TEST_EXP
```

Execute Python tests:
```

> python python/cuspatial/cuspatial/tests/test_geoseries.py

Reviewer 2: Please also show how to run all the pytests.

2. clone the cuSpatial repo

```shell
conda env create -n cuspatial --file conda/environments/all_cuda-118_arch-x86_64.yaml
```

## Build cuSpatial

### From the cuSpatial Dev Container:

Execute `build-cuspatial-cpp to build `libcuspatial`. The following options may be added.
 - `-DBUILD_TESTS=ON`: build `libcuspatial` unit tests.
 - `-DBUILD_BENCHMARKS=ON`: build `libcuspatial` benchmarks.
 - `-DCMAKE_BUILD_TYPE=Debug`: Create a Debug build of `libcuspatial` (default is Release).
In addition, `build-cuspatial-python` to build cuspatial cython components.

### From Bare Metal:

Compile libcuspatial (C++), cuspatial (cython) and C++ tests:
```shell
cd $CUSPATIAL_HOME && \
chmod +x ./build.sh && \
./build.sh libcuspatial cuspatial tests
```
Additionally, the following options are also commonly used:
- `benchmarks`: build libcuspatial benchmarks
- `clean`: remove all existing build artifacts and configuration
Execute `./build.sh -h` for full list of available options.

## Validate Installation with C++ and Python Tests

> - C++ tests are located within the `$CUSPATIAL_HOME/cpp/build/gtests` directory.

Reviewer 2: Where does `build.sh` put the build? For dev container, need latest on path...

```suggestion
- C++ tests are located within the `$CUSPATIAL_HOME/cpp/build/gtests/latest` directory.
```

Author: `latest` should go before `gtests`.

Execute C++ tests:
```shell

> ninja -C $CUSPATIAL_HOME/cpp/build/ test

Reviewer 2: Need `latest` if they are in dev container.
